torch/_tensor.py

```diff
@@ -904,6 +904,55 @@ def rename(self, *names, **rename_map):
         # See Note [rename_ / rename API]
         return update_names(self, names, rename_map, inplace=False)
 
+    def to_sparse_csr(self):
```

Comment:
Notice that if `self` uses `cuda` device, the result of `to_sparse_csr` currently always uses `cpu` device.

Although the CUDA CSR is on the TODO list, this behavior introduces future BC issues.

I suggest changing `to_sparse_csr` implementation to either 
- fail immediately when `self.is_cuda==True`
- or compute all the inputs to `torch.sparse_csr_tensor` such that these use the same device as `self` and let `sparse_csr_tensor` raise the not-implemented exception.

Reply:
Resolved to fail immediately since it would be easier to simply remove this condition once the CUDA backend is available and that extra testing will be needed to make sure that the failure is happening at the right place.